assignment_1/2_kth_element.py

```diff
@@ -0,0 +1,101 @@
+# Implement an algorithm to find the kth to last element of a singly linked list
+# Magdalena Grodzińska
+
+# linked list implementation
+
+class Node:
+    def __init__(self, data, next_node):
+        self.data = data
+        self.next_node = next_node
+
+    def get_data(self):
```

Comment:
Yes, it's based on naming conventions. Code analysis tools (like pylint) will point out code accessing private attributes (that is, attributes with name starting in "_") of foreign objects and code reviewers might scrutinize such code more.

But technically everything is public.